test/benchmarks/run.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+#------------------------------------------------------------------------------
+# Bash script to run optimizer performance tests.
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2022 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -euo pipefail
+
+REPO_ROOT=$(cd "$(dirname "$0")/../../" && pwd)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+
+solc="${SOLIDITY_BUILD_DIR}/solc/solc"
+benchmarks_dir="${REPO_ROOT}/test/benchmarks"
+chains_sol="${benchmarks_dir}/chains.sol"
+time_bin_path=$(which time)
```

Comment:
`which` is kinda weird. It does work in many cases but has quirks and it's really an old csh legacy thing. See [Why not use "which"? What to use then?](https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then) In short, it's safer to use `type` or `command` in Bash.

In this particular case you could use `type -P time`. `-P` tell is to specifically search only `PATH` and ignore functions, aliases, shell keywords, etc.

Alternatively, instead of finding it path you could just do `command time` to run it directly. A bit less explicit but saves you having to define a variable.

Reply:
`time` called from a bash script is a built in function rather than `/usr/bin/time`, and it even says in the man-pages that if users want to use the formatting features, `/usr/bin/time` must be used. calling `command -v time` outputs just `time`, which is the bash built in, but now that you mention it - `command time --format "%e" <program>` does in fact work, so no idea what to think of this.
I didn't know about `-P`, and that actually gives me the behaviour I want, so will update to that.

Reply:
> calling `command -v time` outputs just `time`, which is the bash built in, but now that you mention it - `command time --format "%e" <program>` does in fact work, so no idea what to think of this.

Yeah, that's why I said it's less explicit. I think `command time` works mostly because this bypasses Bash's alias substitutions, keywords. Same as just writing `'time'` (i.e. in quotes). `type -P` seems to be the proper way but I suggested `command time` because it's more succint. I'd use `command time` myself but I'm fine with either as long as it's not `which` :)